### PR TITLE
Add create and update date to task

### PR DESCRIPTION
**Pull request by dominikhajduk**
Need that for reports CR and merge more than appreciated.

### CODE REVIEW

File: taskmanager/features/task.feature [hunk 1]
```diff
@@ -1,26 +1,39 @@
 # language: en
-  Feature: Task management
```
Comment (dominikhajduk) [inline taskmanager/features/task.feature:2]: Formatting is taken from default PHPStorm. In my opinion for simplicity, we should follow them (two spaces for indent).

File: taskmanager/src/Application/Task/Task.php [hunk 2]
```diff
@@ -12,15 +12,19 @@ class Task
...
     /** @var User */
     protected $user;
 
-    public function __construct(string $name, Status $status)
+    public function __construct(string $name, Status $status, ?\DateTime $create = null, ?\DateTime $update = null)
     {
         $this->id = Uuid::uuid4();
         $this->name = $name;
         $this->status = $status;
+        $this->create = $this->setDate($create);
```
Comment (partikus) [inline taskmanager/src/Application/Task/Task.php]:
```
$this->createdAt = new \DateTime()
$this->updatedAt = new \DateTime()
```

File: taskmanager/src/Application/Task/Task.php [hunk 3]
```diff
@@ -41,6 +45,15 @@ public function getName(): string
         return $this->name;
     }
 
+    public function getCreate(): \DateTime
```
Comment (partikus) [inline taskmanager/src/Application/Task/Task.php]: `: \DateTimeInterface`

File: taskmanager/src/Application/Task/Task.php [hunk 4]
```diff
@@ -41,6 +45,15 @@ public function getName(): string
...
+    {
+        return $this->create;
+    }
+
+    public function getUpdate(): \DateTime
```
Comment (partikus) [inline taskmanager/src/Application/Task/Task.php]: idem

File: taskmanager/src/Application/Task/Task.php [hunk 5]
```diff
@@ -60,4 +73,9 @@ public function hasAssignment(): bool
     {
         return $this->user !== null;
     }
+
+    private function setDate(?\DateTime $date): \DateTime
```
Comment (partikus) [inline taskmanager/src/Application/Task/Task.php]: useless

**Review by partikus — CHANGES_REQUESTED** [review]
Good job. Few minor fixes needed. :)

**Comment by dominikhajduk** [comment]
@partikus thanks, improved.

File: taskmanager/src/Application/Task/Task.php [hunk 6]
```diff
@@ -21,6 +25,8 @@ public function __construct(string $name, Status $status)
         $this->id = Uuid::uuid4();
         $this->name = $name;
         $this->status = $status;
+        $this->createdAt = new \DateTime();
+        $this->setUpdatedAt();
```
Comment (partikus) [inline taskmanager/src/Application/Task/Task.php]: maybe sth like `touch(): void`

File: taskmanager/src/Application/Task/Task.php [hunk 7]
```diff
@@ -54,10 +69,16 @@ public function assigned(): User
     public function assign(User $user): void
     {
         $this->user = $user;
+        $this->setUpdatedAt();
```
Comment (partikus) [inline taskmanager/src/Application/Task/Task.php]: idem

File: taskmanager/src/Application/Task/Task.php [hunk 8]
```diff
@@ -54,10 +69,16 @@ public function assigned(): User
...
     }
 
     public function hasAssignment(): bool
     {
         return $this->user !== null;
     }
+
+    private function setUpdatedAt()
```
Comment (partikus) [inline taskmanager/src/Application/Task/Task.php]: `public function touch(): void`

**Review by partikus — CHANGES_REQUESTED** [review]

**Review by tarnawski — DISMISSED** [review]
For me looks nice!

**Review by partikus — APPROVED** [review]

File: taskmanager/src/Application/Task/Task.php (hunk 6 above)
Comment (dominikhajduk) [reply taskmanager/src/Application/Task/Task.php]: @partikus I wasn't sure about `touch` maybe just `update`? Changed in this way. But if you like `touch` I will touch this PR once more :)

Comment (partikus) [reply taskmanager/src/Application/Task/Task.php]: that's fine 👍

**Review by dgrochowski — APPROVED** [review]

**Pull request merged**